ssl_cert_tracker/models.py

```diff
@@ -13,63 +13,93 @@
+    name = models.CharField(max_length=100, blank=True, null=True)
+    bits = models.CharField(max_length=100, blank=True, null=True)
+    md5 = models.CharField(max_length=100, blank=True, null=True)
+    sha1 = models.CharField(max_length=100, blank=True, null=True)
+
+    @classmethod
+    def get_user(cls, username=settings.ORION_USER):
+        """get_user. """
+        user = cls.get_or_create_user(username)
+        return user
+
 
     def update_cert_history(self, pk_val=False):
+        """update_cert_history. """
         if pk_val:
-            orion_id_count = TestCertsHistory.objects.filter(orion_id = self.orion_id).count()
-            print(orion_id_count)
-            if orion_id_count == 0:
-                print ("We are doing insert....")
-                obj = TestCertsHistory(cert_id = int(pk_val), orion_id=self.orion_id, md5=self.md5, status="new",xml_data=self.xml_data )
+            orion_id_count = NmapHistory.objects.filter(orion_id=self.orion_id).count()
+            if orion_id_count == 0: #this is insert
+                obj = NmapHistory(cert_id=int(pk_val), \
+                orion_id=self.orion_id, \
+                md5=self.md5, \
+                status="new", \
+                xml_data=self.xml_data)
                 obj.save()
             else: # this is either update/insert
-                md5_count = TestCertsHistory.objects.filter(orion_id=self.orion_id).filter(md5=self.md5).count()
+                md5_count = NmapHistory.objects.filter(orion_id=self.orion_id).filter(md5=self.md5).count()
                 if md5_count == 0:
-                    print ("We are doing insert....")
-                    obj = TestCertsHistory(cert_id=int(pk_val), \
+                    obj = NmapHistory(cert_id=int(pk_val), \
                                            orion_id=self.orion_id, \
                                            md5=self.md5, \
                                            status="changed", \
                                            xml_data=self.xml_data)
                     obj.save()
                 else:
-                    print ("Certificate found.....We are doing update....")
-                    TestCertsHistory.objects.filter(orion_id=self.orion_id).update(status="found")
+                    NmapHistory.objects.filter(orion_id=self.orion_id).update(status="found")
 
     def save(self, *args, **kwargs):
-        print("We are in Model.save")
-        super(TestCertsData,self).save(*args, **kwargs)
+        super(NmapCertsData, self).save(*args, **kwargs)
         pk_val = self.pk
-        print(pk_val)
-
         self.update_cert_history(pk_val)
+    
+    def get_cert_handle(o_id):
+        """get_cert_handle """
+        return NmapCertsData.objects.get(orion_id = o_id)
+
+    def get_cert_state(o_id, hash_md5):
+        """orion_id_exist """
+        if NmapCertsData.objects.filter(orion_id=o_id).count() == 0:
```

Comment:
if model.objects.count() == 0 can be better expressed as if not model.objects.exists()